rs/replicated_state/src/replicated_state.rs

```diff
@@ -709,6 +720,7 @@ impl ReplicatedState {
             None => {
                 let subnet_id = self.metadata.own_subnet_id.get_ref();
                 if msg.receiver().get_ref() == subnet_id {
+                    // FIXME Assert that this is a request; else check for a matching `Callback`.
```

Comment:
Replaced with a `TODO(MR-523)` because this is part of response deduplication. In addition to canisters, we also need to ensure correctness for subnet queues.